lightning/src/routing/router.rs

```diff
@@ -463,6 +463,13 @@ struct PathBuildingHop<'a> {
 	value_contribution_msat: u64,
 }
 
+impl<'a> core::fmt::Debug for PathBuildingHop<'a> {
+	fn fmt(&self, f: &mut core::fmt::Formatter) -> Result<(), core::fmt::Error> {
+        write!(f, "PathBuildingHop: node_id: {:?}, short_channel_id: {}, fee_msat: {}, path_penalty_msat: {}, path_htlc_minimum_msat: {}, cltv_expiry_delta: {}", self.node_id, self.candidate.short_channel_id(), self.fee_msat, self.path_penalty_msat, self.path_htlc_minimum_msat, self.candidate.cltv_expiry_delta())?;
```

Comment:
You can use `Formatter `methods as shown here: https://doc.rust-lang.org/std/fmt/struct.DebugStruct.html#examples

That way pretty-printing is honored if the caller specifies it.